Assets/Scripts/Framework/Gameplay/MiniGames/Objects/TurningEnemy.cs

```diff
@@ -0,0 +1,69 @@
+using UnityEngine.Events;
+using UnityEngine;
+using TMPro;
+
+namespace Framework.Gameplay.MiniGames.Objects
+{
+    public sealed class TurningEnemy : MonoBehaviour
+    {
+        [SerializeField] private UnityEvent onCaught = new();
+        [HideInInspector] public bool minigameActive;
+        [SerializeField] private bool _isWatching;
+        private SpriteRenderer _spriteRenderer; //temporary until art implementation
+        private float _lookInterval;
+        private Timer timer;
+
+        private void Start() 
+        {
+            _spriteRenderer = GetComponent<SpriteRenderer>();
+            timer = GetComponent<Timer>(); 
+        }
+
+        public void OnMiniGameActive()
+        {
+            minigameActive = true;
+            SetTimer();
+        }
+
+        public void OnMinigameDeactivate()
+        {
+            minigameActive = false;
+            timer.SetCanCount(false);
+        }
+
+        public void RoundBehaviour()
+        {
+            timer.ResetTimer();
+            Turn();
+            SetTimer();
+        }
+
+        /// <summary>
+        /// shows when the character is turned.( the color is temporary until art implementation)
+        /// </summary>
+        private void Turn()
+        {
+            _isWatching = !_isWatching;
+            if (_isWatching )
+                _spriteRenderer.color = Color.red;
+            else
+                _spriteRenderer.color = Color.green;
+        }
+
+        private void SetTimer()
+        {
+            _lookInterval = Random.Range(2, 4);
```

Comment:
Deze 2 en 4 liever een SerializeField van maken. Kan als 2 floats of als 1 Vector2 met mooie naam, als je dat doe is er ook een `RangeVector2` attribute.